### PR TITLE
Used luxon instead of moment #1vy5ryh

### PR DESCRIPTION
**Pull request by disha1202**

### CODE REVIEW

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/TimezoneModal.vue [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  name: "TimeZoneModal",
+  data() {
+    return {
+      queryString: '',
+      filteredTimeZones: [],
+      timeZones: [],
+      timeZoneId: ''
+    }
+  },
+  computed: {
```
Comment (ymaheshwari1) [inline src/views/TimezoneModal.vue:77]: Removed the computed block as its not needed.

File: src/views/TimezoneModal.vue [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    closeModal() {
+      modalController.dismiss({ dismissed: true });
+    },
+    async saveAlert() {
+      const message = this.$t("Are you sure you want to change the time zone to?", { timeZoneId: this.timeZoneId });
+      const alert = await alertController.create({
+        header: this.$t("Update time zone"),
+        message,
+        buttons: [
+            {
```
Comment (ymaheshwari1) [inline src/views/TimezoneModal.vue:89]: Improve indentation

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/App.vue [hunk 3]
```diff
@@ -23,6 +24,29 @@ export default defineComponent({
     }
   },
   methods: {
+    async timeZoneDifferentAlert(payload: any) {
+      const alert = await alertController.create({
+        header: this.$t("Change time zone"),
+        message: this.$t('Would you like to update your time zone to . Your profile is currently set to . This setting can always be changed from the settings menu.', { localTimeZone: payload.localTimeZone, profileTimeZone: payload.profileTimeZone }),
+        buttons: [
+            {
```
[the remark this answers is not in the document]
Comment (disha1202) [reply src/App.vue]: Done sir

File: src/views/TimezoneModal.vue (hunk 1 above)
Comment (disha1202) [reply src/views/TimezoneModal.vue]: Removed.

File: src/views/TimezoneModal.vue (hunk 2 above)
Comment (disha1202) [reply src/views/TimezoneModal.vue]: Improved indentation

File: src/views/TimezoneModal.vue [hunk 4]
```diff
@@ -120,9 +116,10 @@ export default defineComponent({
           // Currently backend API returns some of the legacy timezones which are not found in moment list
           // Due to which if we set them we get an error
           // Filtered them out till it is fixed at backend
-          this.timeZones = resp.data.filter((timeZone: any) => {
```
Comment (adityasharma7) [inline src/views/TimezoneModal.vue:123]: Why we have commented this out?

File: src/App.vue [hunk 5]
```diff
@@ -29,20 +29,20 @@ export default defineComponent({
         header: this.$t("Change time zone"),
         message: this.$t('Would you like to update your time zone to . Your profile is currently set to . This setting can always be changed from the settings menu.', { localTimeZone: payload.localTimeZone, profileTimeZone: payload.profileTimeZone }),
         buttons: [
-            {
-              text: this.$t("Dismiss"),
-              role: 'cancel',
-              cssClass: 'secondary'
+          {
```
Comment (adityasharma7) [inline src/App.vue:32]: Please revert all the unrelated indentation changes

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/TimezoneModal.vue (hunk 4 above)
Comment (disha1202) [reply src/views/TimezoneModal.vue:123]: Removed the commented code and comments because we are not using moment anymore.

File: src/App.vue (hunk 5 above)
Comment (disha1202) [reply src/App.vue:32]: Reverted indentation changes.

File: src/views/TimezoneModal.vue (hunk 4 above)
Comment (adityasharma7) [reply src/views/TimezoneModal.vue:123]: I think here we are checking if the server timezone belongs to moment as well. We should also handle it for luxon. Check if we are similar method in Luxon

File: src/views/TimezoneModal.vue [hunk 6]
```diff
@@ -88,16 +84,16 @@ export default defineComponent({
...
-              text: this.$t("Cancel"),
-            },
-            {
-              text: this.$t("Confirm"),
-              handler: () => {
-                this.setUserTimeZone();
-                }
-              }
-            ],
+          {
```
Comment (adityasharma7) [inline src/views/TimezoneModal.vue:87]: Revert unrelated indentation changes

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/TimezoneModal.vue (hunk 4 above)
Comment (disha1202) [reply src/views/TimezoneModal.vue:123]: Handled it for luxon as well

File: src/views/TimezoneModal.vue (hunk 6 above)
Comment (disha1202) [reply src/views/TimezoneModal.vue]: Reverted unrelated indentation changes

**Review by adityasharma7 — APPROVED** [review]
